### PR TITLE
Latest Version Fix & Connect cleanup

### PR DESCRIPTION
**Pull request by gaelder**
- Properly implemented determination of version when the api receives 'latest' as the version request
- Reworked connect to be in a single location and to cache a connection promise rather than create endless ones

### CODE REVIEW

File: src/server/database/utilities.js [hunk 1]
```diff
@@ -2,6 +2,16 @@
 const r = require('rethinkdb');
 const config = require('./config');
 
+let connectionPromise = null;
+
+function connect() {
```
Comment (d2fong) [inline src/server/database/utilities.js]: @maxkfranz Do you think it makes sense to use lodash.memoize here?

Comment (d2fong) [reply src/server/database/utilities.js]: This is about caching the connection to rethinkdb.

Comment (maxkfranz) [reply src/server/database/utilities.js]: Yes, that would work

Comment (d2fong) [reply src/server/database/utilities.js]: @gaelder learn about memoize https://lodash.com/docs/4.17.4#memoize

And use that to cache the rethinkdb connection.

Comment (gaelder) [reply src/server/database/utilities.js]: @d2fong Done. Let me know if I did something weird.

**Pull request merged**